operator/cmd/sidecar/sidecar.go

```diff
@@ -0,0 +1,146 @@
+
+func Command() *cobra.Command {
+	var (
+		metricsAddr                string
+		probeAddr                  string
+		pprofAddr                  string
+		clusterNamespace           string
+		clusterName                string
+		decommissionRequeueTimeout time.Duration
+		decommissionVoteInterval   time.Duration
+		decommissionMaxVoteCount   int
+		redpandaYAMLPath           string
+	)
+
+	cmd := &cobra.Command{
+		Use:   "run",
+		Short: "Run the redpanda sidecar",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+
+			return Run(
+				ctx,
+				metricsAddr,
+				probeAddr,
+				pprofAddr,
+				clusterNamespace,
+				clusterName,
+				decommissionRequeueTimeout,
+				decommissionVoteInterval,
+				decommissionMaxVoteCount,
+				redpandaYAMLPath,
+			)
+		},
+	}
+
+	cmd.Flags().StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	cmd.Flags().StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	cmd.Flags().StringVar(&pprofAddr, "pprof-bind-address", ":8082", "The address the metric endpoint binds to.")
+	cmd.Flags().StringVar(&clusterNamespace, "cluster-namespace", "", "The namespace of the cluster that this sidecar manages.")
+	cmd.Flags().StringVar(&clusterName, "cluster-name", "", "The name of the cluster that this sidecar manages.")
+	cmd.Flags().DurationVar(&decommissionRequeueTimeout, "decommission-requeue-timeout", 10*time.Second, "The time period to wait before rechecking a broker that is being decommissioned.")
+	cmd.Flags().DurationVar(&decommissionVoteInterval, "decommission-vote-interval", 30*time.Second, "The time period between incrementing decommission vote counts since the last decommission conditions were met.")
+	cmd.Flags().IntVar(&decommissionMaxVoteCount, "decommission-vote-count", 2, "The number of times that a vote must be tallied when a resource meets decommission conditions for it to actually be decommissioned.")
+	cmd.Flags().StringVar(&redpandaYAMLPath, "redpanda-yaml", "/etc/redpanda/redpanda.yaml", "Path to redpanda.yaml")
+
+	return cmd
+}
+
+func Run(
+	ctx context.Context,
+	metricsAddr string,
+	probeAddr string,
+	pprofAddr string,
+	clusterNamespace string,
+	clusterName string,
+	decommissionRequeueTimeout time.Duration,
+	decommissionVoteInterval time.Duration,
+	decommissionMaxVoteCount int,
+	redpandaYAMLPath string,
+) error {
+	setupLog := ctrl.LoggerFrom(ctx).WithName("setup")
+
+	if clusterNamespace == "" {
+		err := errors.New("must specify a cluster-namespace parameter")
+		setupLog.Error(err, "no cluster namespace provided")
+		return err
+	}
+
+	if clusterName == "" {
+		err := errors.New("must specify a cluster-name parameter")
+		setupLog.Error(err, "no cluster name provided")
+		return err
+	}
+
+	scheme := runtime.NewScheme()
+
+	for _, fn := range schemes {
+		utilruntime.Must(fn(scheme))
+	}
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
```

Comment:
Technically we'll need another set of RBAC permissions for leader election. `run.go` currently has markers set in the file. Should we copy them over here?